unit_test/build_selector_test.py

```diff
@@ -69,3 +71,20 @@ def test_build_braces():
     assert build_selector('cp37-manylinux1_x86_64')
     assert not build_selector('cp38-manylinux1_x86_64')
     assert not build_selector('cp39-manylinux1_x86_64')
+
+
+def test_build_limited_python():
+    build_selector = BuildSelector(build_config="*", skip_config="", requires_python=SpecifierSet(">=3.6"))
+
+    assert not build_selector('cp27-manylinux1_x86_64')
+    assert build_selector('cp36-manylinux1_x86_64')
+    assert build_selector('cp37-manylinux1_x86_64')
+    assert not build_selector('cp27-manylinux1_i686')
+    assert build_selector('cp36-manylinux1_i686')
+    assert build_selector('cp37-manylinux1_i686')
+    assert not build_selector('cp27-win32')
+    assert build_selector('cp36-win32')
+    assert build_selector('cp37-win32')
+    assert not build_selector('pp27-win32')
+    assert build_selector('pp36-win32')
+    assert build_selector('pp37-win32')
```

Comment:
Another test with something more exotic like `SpecifierSet(">=2.7.0, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, !=3.5.*")` might be valuable. Also something with a patch version, like `SpecifierSet(">=2.7.9")`.

Reply:
I've never seen a patch version here, but there's no reason it wouldn't be valid. I've been wanting got add minor versions to the macOS identifiers; it would have made the generator script simpler and would have been more consistent (even though we currently never need more than the first two digits. Actually, if we include packaging now, we could keep them Version's and tidy up a bit.

Reply:
I understand including it for completeness, but a wheel tag does not include the patch version of Python it was built on; a wheel built with 3.9.1 can be loaded on 3.9.0. So I'm not sure how putting a patch version in your Python-Requires could ever be useful. Honestly, it might be better to always pretend to be on the ".0" patch version, so mistaken identifiers like `>=2.7, !=3.0, !=3.1, !=3.2, !=3.3, !=3.4, !=3.5` would still work (I'd exactly match pip here; I'd have to see what it does; I'd assume it checks the full version).

Reply:
Python-Requires tells you if the current version of Python is "good enough" to be used; it does not tell you to go download a specific version like an install requirement could. Since Python does not add new features in patch releases (ignore 2.7), you are not likely to "drop" support for a patch version, there's no reason to put a patch version in here. (pip would go back and load an older version of your software that did support a patch version). How about, for now, we add `.99` to any two digit version when checking this? Eventually we could try to compute an actual version, but there really isn't an answer for PyPy, and Linux will depend on the image, and you'd need to load it to detect what Python's where in it.  `>=3.6.3` and `!=3.6.6` would both correctly work if we ask, when picking the identifiers that match, if `3.6.99` matches.

As I've said before, this is about the Python language version, not the CPython version. And the language doesn't have patch versions. So I'd treat placing a patch version in Python-Requires here more as a user error.

Reply:
> this is about the Python language version, not the CPython version. And the language doesn't have patch versions. 

This seems like the crux to me. Agreed. I'd be curious what PyPy does for the patch version, do they just put `0`? Anyway, the `.99` hack seems to make the most sense to me.

Reply:
I bet these values are the ones used: https://www.pypy.org/compat.html

Reply:
And here's the value for 3.6: https://morepypy.blogspot.com/2020/11/pypy-733-triple-release-python-37-36.html

Reply:
> As I've said before, this is about the Python language version, not the CPython version. And the language doesn't have patch versions. So I'd treat placing a patch version in Python-Requires here more as a user error.

It shouldn't, no. But in practice, the only real existing standard for Python seems to be the CPython reference implementation (just ask PyPy how much they like this ;-) ). It's very rare/almost non-existant that the language itself changes, though, in patch releases, as I guess that's still not very semantically versioned.

Reply:
The underlying question was resolved, but I still think-

> Another test with something more exotic like SpecifierSet(">=2.7.0, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, !=3.5.*") might be valuable. Also something with a patch version, like SpecifierSet(">=2.7.9").

Reply:
I could have sworn I added this...